src/ashell/ide-comms.c

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2016-2017, Intel Corporation.
+
+/**
+ * @file
+ * @brief Device-side service for Web IDE message exchanges based on WebUSB.
+ *
+ */
+
+// C includes
+#include <ctype.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+// ZJS includes
+#include "ashell.h"
+#include "ide-comms.h"
+
+// Process a buffer (part of a message) in WebUSB driver context.
+void ide_receive(u8_t *buffer, size_t len)
+{
+    static char rx_buffer[P_SPOOL_SIZE + 1];
```

Comment:
I don't like the idea of having static buffers in ashell as it means that memory is always held up, even when we aren't sending / receiving data from webUSB.  @grgustaf any opinions on the matter?

Reply:
Throughout zephyr there are a lot of places static buffers are used and held up and I agree this is wrong. In this particular case this is here only to make the _current_ zephyrjs-ide work with the IDE protocol, since the ide is sending each character separately at the moment. This will go away when WiP flag is removed.